askama_shared/src/generator.rs

```diff
@@ -211,6 +215,53 @@ impl<'a, S: std::hash::BuildHasher> Generator<'a, S> {
         buf.writeln("}")
     }
 
+    // Implement Tests
+    fn impl_tests(&mut self, buf: &mut Buffer) -> Result<(), CompileError> {
+        buf.writeln(&format!(
+            "#[cfg(test)] mod __{}_tests_generated {{",
+            self.input.ast.ident.to_string().to_lowercase()
+        ))?;
+        // TODO
+        //if cfg!(feature = "with-i18n") {
+        self.impl_i18n_tests(buf)?;
+        //}
+
+        buf.writeln("}")
+    }
+
+    fn impl_i18n_tests(&mut self, buf: &mut Buffer) -> Result<(), CompileError> {
+        let messages = &self.localized_messages;
+        if messages.len() > 0 {
+            let loc_ty = self.input.localizer.as_ref().unwrap().1;
+            let ast = (quote! {
+
+
+                #[test]
+                fn test_i18n_default_coverage() {
+                    let messages = &[
+                        #(#messages),*
+                    ][..];
+    
+                    // create default localizer
+                    let localizer = super::#loc_ty::default();
```

Comment:
Here we need to be able to somehow access the static loader. Currently an object of the generated struct type is used for that.